Content.Server/Xenoarchaeology/Equipment/Systems/ArtifactCrusherSystem.cs

```diff
@@ -0,0 +1,135 @@
+using Content.Server.Body.Systems;
+using Content.Server.Power.Components;
+using Content.Server.Power.EntitySystems;
+using Content.Server.Stack;
+using Content.Server.Storage.Components;
+using Content.Server.Xenoarchaeology.XenoArtifacts;
+using Content.Shared.Body.Components;
+using Content.Shared.Damage;
+using Content.Shared.Verbs;
+using Content.Shared.Xenoarchaeology.Equipment;
+using Robust.Shared.Collections;
+using Robust.Shared.Random;
+using Robust.Shared.Timing;
+
+namespace Content.Server.Xenoarchaeology.Equipment.Systems;
+
+/// <inheritdoc/>
+public sealed class ArtifactCrusherSystem : SharedArtifactCrusherSystem
+{
+    [Dependency] private readonly IGameTiming _timing = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly ArtifactSystem _artifact = default!;
+    [Dependency] private readonly BodySystem _body = default!;
+    [Dependency] private readonly DamageableSystem _damageable = default!;
+    [Dependency] private readonly StackSystem _stack = default!;
+
+    /// <inheritdoc/>
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<ArtifactCrusherComponent, GetVerbsEvent<AlternativeVerb>>(OnGetVerbs);
+        SubscribeLocalEvent<ArtifactCrusherComponent, PowerChangedEvent>(OnPowerChanged);
+    }
+
+    private void OnGetVerbs(Entity<ArtifactCrusherComponent> ent, ref GetVerbsEvent<AlternativeVerb> args)
+    {
+        if (!args.CanAccess || !args.CanInteract || args.Hands == null || ent.Comp.Crushing)
+            return;
+
+        if (!TryComp<EntityStorageComponent>(ent, out var entityStorageComp) || entityStorageComp.Contents.ContainedEntities.Count == 0)
+            return;
+
+        if (entityStorageComp.Contents.Contains(args.User) || !this.IsPowered(ent, EntityManager))
+            return;
+
+        var verb = new AlternativeVerb
+        {
+            Text = Loc.GetString("artifact-crusher-verb-start-crushing"),
+            Priority = 2,
+            Act = () => StartCrushing((ent, ent.Comp, entityStorageComp))
+        };
+        args.Verbs.Add(verb);
+    }
+
+    private void OnPowerChanged(Entity<ArtifactCrusherComponent> ent, ref PowerChangedEvent args)
+    {
+        if (!args.Powered)
+            StopCrushing(ent);
+    }
+
+    public void StartCrushing(Entity<ArtifactCrusherComponent, EntityStorageComponent> ent)
+    {
+        var (_, crusher, _) = ent;
+
+        if (crusher.Crushing)
+            return;
+
+        crusher.Crushing = true;
+        crusher.NextSecond = _timing.CurTime + TimeSpan.FromSeconds(1);
+        crusher.CrushEndTime = _timing.CurTime + crusher.CrushDuration;
```

Comment:
If CrushDuration is < 1s wouldn't that cause it to just end?

Reply:
the NextSecond tracker and CrushEndTime are used for two different things. NextSecond is just for doing the crushing damage every second and the CrushEndTime is tracked entirely separately for seeing when the final "crush" will happen.

I'm not sure what your comment means but it would work fine with a sub-one second crush time.